app/services/induction/change_mentor.rb

```diff
@@ -7,6 +7,7 @@ def call
                                             changes: { mentor_profile: })
 
       induction_record.participant_profile.update!(mentor_profile:)
+      mentor_profile&.user&.touch
```

Comment:
Would it be better to encapsulate this in the `InductionRecord` model as above and touch the user through lifecycle callbacks to cover other use cases? doing it this way means if the `induction_record.mentor_profile` is updated elsewhere it won't touch the user?

Reply:
We usually read the `mentor_profile_id` from induction records rather than the profile or always should be, although checking the repository seems like a mix due to old code/habits. So ideally we'll need to add callbacks to both IRs and profiles. Adding a callback on induction records will become too noisy as we always create new induction records with mentor profile id populated (see https://github.com/DFE-Digital/early-careers-framework/blob/main/app/services/induction/change_mentor.rb#L6) rather than update in place, which could accidentally catch any new induction record created for other reasons.

I think adding touching the user in explicit cases when changing the mentor as it was in [app/services/induction/change_mentor.rb](https://github.com/DFE-Digital/early-careers-framework/pull/4271/files#diff-1fb68b60719d16f750323df92b961b0d719f270fc50a15bba53297bd8f6abfee) would be preferable to avoid any unexpected behaviour. For instance we already have some strange syncs here which appear cyclical: https://github.com/DFE-Digital/early-careers-framework/blob/main/app/models/participant_profile/ecf.rb#L149 and will catch us off guard. Happy to chat after standup tomorrow to go through the touch model scenarios